gui/gui/src/TRootGuiFactory.cxx

```diff
@@ -49,17 +51,48 @@ TApplicationImp *TRootGuiFactory::CreateApplicationImp(const char *classname,
    TRootApplication *app = new TRootApplication(classname, argc, argv);
    if (!app->Client()) {
       delete app;
-      app = 0;
+      app = nullptr;
    }
    return app;
 }
 
+////////////////////////////////////////////////////////////////////////////////////////
+/// Show warning that TWebCanvas will be started by default
+
+void TRootGuiFactory::ShowWebCanvasWarning()
+{
+   static bool show_warn = true;
+   if (!show_warn || gROOT->IsWebDisplay()) return;
+   show_warn = false;
+
+   std::cout << "\n"
+                "   !!! ATTENTION !!! \n"
+                "\n"
+                "ROOT comes with a web-based canvas, which is now being started. \n"
+                "Revert to default by setting \"Canvas.Name: TRootCanvas\" in rootrc file or\n"
+                "by starting \"root --web=off\". Web-based TCanvas can be used in batch mode for\n"
+                "image generation when running with \"root -b --web\"\n"
+                "Find more info on https://root.cern/for_developers/root7/#twebcanvas\n"
+                "\n";
+}
+
 ////////////////////////////////////////////////////////////////////////////////
 /// Create a ROOT native GUI version of TCanvasImp
 
 TCanvasImp *TRootGuiFactory::CreateCanvasImp(TCanvas *c, const char *title,
                                              UInt_t width, UInt_t height)
 {
+   TString canvName = gEnv->GetValue("Canvas.Name", "TWebCanvas");
+   if (canvName == "TWebCanvas") {
+      auto ph = gROOT->GetPluginManager()->FindHandler("TCanvasImp", "TWebCanvas");
+
+      if (ph && ph->LoadPlugin() != -1) {
+         ShowWebCanvasWarning();
+         auto imp = (TCanvasImp *) ph->ExecPlugin(6, c, title, 0, 0, width, height);
+         if (imp) return imp;
```

Comment:
and `if (!imp)` we should probably issue a warning?

Reply:
It is code template used everywhere in ROOT. In very few places one shows warnings. 
I could provide separate PR which makes warnings for all gui/gpad classes. 
There it fully missing